cli/flox/src/commands/install.rs

```diff
@@ -228,23 +226,16 @@ impl Install {
         let mut environment = concrete_environment.into_dyn_environment();
 
         // Get a list of the packages that this environment is already overriding via composition.
-        let lockfile_path = environment.lockfile_path(&flox)?;
-        let canonicalized_lockfile_path_res = CanonicalPath::new(lockfile_path);
-        let existing_composer_package_overrides = if let Ok(path) = canonicalized_lockfile_path_res
-        {
-            let lockfile_content =
-                std::fs::read_to_string(&path).context("Failed to read lockfile")?;
-            let lockfile =
-                Lockfile::from_str(&lockfile_content).context("Failed to parse lockfile")?;
-            let existing_composer_overrides = lockfile
+        let maybe_lockfile = environment.existing_lockfile()?;
+        let existing_composer_package_overrides = if let Some(lockfile) = maybe_lockfile {
```

Comment:
This could also be done by calling `map` on `maybe_lockfile`, but I think it's marginally clearer this way since there would be nested `map`s otherwise.